### PR TITLE
Return more descriptive error when dimensions can't be expanded

### PR DESCRIPTION
**Pull request by starstuck**
This is pull request cherry-picked from changes by @davecromberge. It pretty much addresses issue #43.

### CODE REVIEW

File: src/dql_resolution.erl [hunk 1]
```diff
@@ -8,7 +8,9 @@
 %%%-------------------------------------------------------------------
 -module(dql_resolution).
 
--export([resolve/2, propagate/1, start_time/1]).
+% -export([resolve/2, propagate/1, start_time/1]).
+
+-compile([export_all]).
```
Comment (Licenser) [inline src/dql_resolution.erl]: I'm not a fan of export all, specific exports serve a reason of declaring a clear interface and making looking up functions easier.

**Review by Licenser — COMMENTED** [review]
just one little thing otherwise looks great :D

File: src/dql_resolution.erl (hunk 1 above)
Comment (davecromberge) [reply src/dql_resolution.erl]: This was my mistake, I forgot to remove this as I was testing the changes through the shell

Comment (Licenser) [reply src/dql_resolution.erl]: Cool!

handy trick:
```erlang
-ifdef(TEST).
-compile([export_all]).
-endif.
```
that way setting the erlang env TEST allows you to export all and test w/o poluting the exports during development/runtime :)

Comment (davecromberge) [reply src/dql_resolution.erl]: That's neat, I thought it was only for eunit tests and didn't know it applied to the shell as well

Comment (Licenser) [reply src/dql_resolution.erl]: not automatically but you can set it in the rebar.config for example makeing a test-shell profile

**Comment by starstuck** [comment]
It turned out, that those extra exports are not used in tests, so I just removed it.

**Comment by Licenser** [comment]
yay!

**Comment by Licenser** [comment]
tagged as `0.4.20`

**Comment by starstuck** [comment]
Cheers!

**Pull request merged**